test/functional/tests/io_class/test_io_class_eviction_pinned_after_unpin.py

```diff
@@ -0,0 +1,136 @@
+        ioclass_config.add_ioclass(*str(IoClass.default(allocation="0.00")).split(","))
+
+    with TestRun.step("Add IoClasses for dirs"):
+        pinned_io_class = io_classes[0]
+        _add_and_load_io_classes(cache.cache_id, io_classes)
+
+    with TestRun.step("Reset cache stats"):
+        _refresh(cache)
+
+    with TestRun.step("Check occupancy before fio"):
+        occupancy_before = get_io_class_occupancy(cache, pinned_io_class.id)
+        if occupancy_before.get_value() != 0:
+            TestRun.fail(
+                f"Incorrect inital occupancy for pinned ioclass: {pinned_io_class.id}."
+                f" Expected 0, got: {occupancy_before}"
+            )
+
+    with TestRun.step("Run fio on IO classes"):
+        for io_class in io_classes:
+            run_io_dir(f"{io_class.dir_path}/test_file", int(cache_line_count / Unit.Blocks4096))
+
+    with TestRun.step("Check if eviction hasn't occured on pinned IO class"):
+        occupancy_after = get_io_class_occupancy(cache, pinned_io_class.id, percent=True)
+        if not isclose(occupancy_after, pinned_io_class.max_occupancy * 100, rel_tol=0.05):
+            TestRun.fail(
+                f"""
+                Pinned class has not been fully filled 
+                Expected: {pinned_io_class.max_occupancy * 100} Actual: {occupancy_after}
+                """
+            )
+
+    with TestRun.step("Unpin ioclass to and set its priority to be equal second IO class"):
+        io_classes = [
+            IoclassConfig(1, 0, 1, f"{mountpoint}/A"),
+            IoclassConfig(2, 0, 1, f"{mountpoint}/B"),
+        ]
+        _add_and_load_io_classes(cache.cache_id, io_classes)
+
+    with TestRun.step("Reset cache stats again"):
+        _refresh(cache)
+
+    with TestRun.step("Run fio on was pinned IO class"):
+        run_io_dir(f"{pinned_io_class.dir_path}/test_file", int(cache_line_count / Unit.Blocks4096))
+        pinned_occupancy = get_io_class_occupancy(cache, pinned_io_class.id, percent=True)
+        if not isclose(pinned_occupancy, pinned_io_class.max_occupancy * 100, rel_tol=0.05):
+            TestRun.fail(
+                f"""
+                Pinned class has not been fully filled 
+                Expected: {pinned_io_class.max_occupancy * 100} Actual: {occupancy_after}
+                """
+            )
+    with TestRun.step("Run fio on lower priority io ioclass "):
+        run_io_dir(f"{io_classes[1].dir_path}/test_file", int(cache_line_count / Unit.Blocks4096))
+
+    with TestRun.step("Check if data from 'was pinned' IO class was evicted"):
+        occupancy_after_change = get_io_class_occupancy(cache, pinned_io_class.id, percent=True)
+        if isclose(occupancy_after, occupancy_after_change, rel_tol=0.05):
+            TestRun.fail(
+                f"""
+                Data was not evicted: before:  {occupancy_after}
+                after: {occupancy_after_change}
+                """
+            )
+
+
+def _refresh(cache):
```

Comment:
I suggest using a more meaningful name. The current one doesn't really say what is being refreshed